### PR TITLE
#63 fix success response should be json object rule

### LINKED ISSUE #63: Incorrect 'Response As JSON Object:' violation
Given the following swagger spec:

```yaml
swagger: '2.0'

info:
  version: "0.0.1"
  title: API Example

schemes:
    - "https"
basePath: /
produces:
  - application/json
consumes:
  - application/json

securityDefinitions:
  oauth2:
    type: oauth2
    flow: password
    tokenUrl: https://tokeninfo.example.com
    scopes:
      uid: Unique identifier of the user accessing the service.

security:
  - oauth2: [uid]

paths:
  '/resources':
    get:
      summary: List all resources
      responses:
        200:
          description: List of all resources.
          schema:
            type: object
            properties:
              items:
                type: array
                items:
                    '$ref': '#/definitions/Resource'
        500:
          description: Unexpected error

definitions:
  Resource:
    type: object
    properties:
      name:
        type: string
```

I get this violation when running the CLI:

```
$ cli/bin/zally swagger.yaml
Found the following MUST violations
===================================
Response As JSON Object:
	Always Return JSON Objects As Top-Level Data Structures To Support Extensibility
	/resources/GET/200

Summary:
=======
COULD violations: 0
SHOULD violations: 0
MUST violations: 1
```

Since the response type of `/resources/GET/200` is:

```json
{
  "items": [Resource,]
}
```

I would not expect it to be a violation.

### PR DESCRIPTION
**Pull request by mkulak**
Closes #63

Needs additional fixes (for now one of the tests fails).

### CODE REVIEW

**Comment by mkulak** [comment]
Unfortunately I don't have enough time to finish this in nearest future. Maybe @ishishov will continue?

File: server/src/main/java/de/zalando/zally/rules/SuccessResponseAsJsonObjectRule.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import io.swagger.models.ModelImpl
+import io.swagger.models.Swagger
+import io.swagger.models.properties.Property
+import io.swagger.models.properties.RefProperty
+import org.springframework.stereotype.Component
+import java.util.*
+
+
+@Component
+open class SuccessResponseAsJsonObjectRule : Rule {
```
Comment (fmueller) [inline server/src/main/java/de/zalando/zally/rules/SuccessResponseAsJsonObjectRule.kt]: What was the reason for you to open this class for inheritance by annotating it with `open` keyword?

**Comment by fmueller** [comment]
👍

**Comment by netme** [comment]
👍

**Comment by netme** [comment]
Restarted tests

**Comment by netme** [comment]
If tests will pass, I will merge it ignoring "This branch is out-of-date with the base branch" message.

**Pull request merged**